--- a/userGuestbook.tpl	Mon Mar  7 18:40:42 2011
+++ b/userGuestbook.tpl	Tue Jun 26 12:21:21 2012
@@ -56,9 +56,16 @@
 						//]]>
 					</script>
 					
+					{assign var='canReadDeletedEntry' value=false}
+					{if ($this->user->userID == $user->userID) || $this->user->getPermission('mod.guestbook.canDeletedEntryCompletely')}
+						{assign var='canReadDeletedEntry' value=true}
+					{/if}
+					
+					{if !$entry->isDeleted || $canReadDeletedEntry}
+					<div class="messageWrapper">
 					<div class="message">
 						<div class="messageInner {if $entry->commentTime > 0}{cycle advance=false values='container-1,container-2'}{else}{cycle values='container-1,container-2'}{/if}">
+							{if !$entry->isDeleted}
							<a id="entry{@$entry->entryID}"></a>
+							{/if}
 							<div class="messageHeader">
 								<p class="messageCount">
 									<a href="index.php?page=UserGuestbook&amp;userID={@$userID}&amp;entryID={@$entry->entryID}{@SID_ARG_2ND}#entry{@$entry->entryID}" title="{lang}wcf.user.guestbook.entry.permalink{/lang}" class="messageNumber">{#$messageNumber}</a>
@@ -146,6 +153,24 @@
 							</div>
 						</div>
 					{/if}
+					</div>
+					{/if}
+					
+					{if $entry->isDeleted && $canReadDeletedEntry}
+						{cycle name=entryCycle print=false}
+						<div class="message messageMinimized" id="hiddenEntryInfo{@$entry->entryID}">
+							<div class="messageInner container-{cycle name=entryCycle}">
+								<a id="entry{@$entry->entryID}"></a>
+								<img src="{icon}postTrashM.png{/icon}" alt="" />
+								<p class="userCredits smallFont light">
+									<a onclick="showContent('entry{@$entry->entryID}', 'hiddenEntryInfo{@$entry->entryID}')" title="{lang}wcf.user.guestbook.showEntry{/lang}">
+										<span>{lang}wcf.user.guestbook.deletedEntry{/lang}</span>
+									</a>
+								</p>
+							</div>
+						</div>
+					{/if}
+					
 					{assign var='messageNumber' value=$messageNumber-1}
 				{/foreach}
 			</div>
